src/middlewared/middlewared/plugins/apps/resources.py

```diff
@@ -112,6 +112,29 @@ async def ip_choices(self):
             for ip in await self.middleware.call('interface.ip_in_use', {'static': True, 'any': True})
         }
 
+    @accepts(roles=['CATALOG_READ'])
+    @returns(Dict(
+        'available_space',
+        additional_attrs=True,
+        example={
+            'parsed': 21289574400,
+            'rawvalue': '21289574400',
+            'value': '19.8G',
+            'source': 'NONE',
+            'source_info': None,
+        }
+    ))
+    async def available_space(self):
+        """
+        Returns space available in the configured apps pool which apps can consume
+        """
+        await self.middleware.call('docker.state.validate')
+        return (await self.middleware.call(
```

Comment:
We need to get this data using a less expensive operation. We're using the process pool here when I fear the UI team is going to call this endpoint  _a lot_. I'd rather do `filesystem.statfs`